src/libs/API/types.ts

```diff
@@ -0,0 +1,312 @@
+import type {ValueOf} from 'type-fest';
+import type CONST from '@src/CONST';
+import type {EmptyObject} from '@src/types/utils/EmptyObject';
+import type * as Parameters from './parameters';
+import type SignInUserParams from './parameters/SignInUserParams';
+import type UpdateBeneficialOwnersForBankAccountParams from './parameters/UpdateBeneficialOwnersForBankAccountParams';
+
+type ApiRequestWithSideEffects = ValueOf<typeof CONST.API_REQUEST_TYPE, 'MAKE_REQUEST_WITH_SIDE_EFFECTS' | 'READ'>;
```

Comment:
```suggestion
type ApiRequest = ValueOf<typeof CONST.API_REQUEST_TYPE>;
```

This is enough as it includes all the three types.

Reply:
`ApiRequestWithSideEffects` is used in `makeRequestWithSideEffects`, I deliberately didn't include `write`

From my research `makeRequestWithSideEffects` is never used for `apiRequestType === write`, so I limited it to just `read | makeRequestWithSideEffects`. Is there any reason to include `write` commands as well?

Reply:
`makeRequestWithSideEffects` supports all three types (based on the comments) it just happens that we are not using it for any write command at the moment